domain/relation/state/relation.go

```diff
@@ -393,18 +393,45 @@ func (st *State) GetAllRelationDetails(ctx context.Context) ([]domainrelation.Re
 
 	var relationsDetails []domainrelation.RelationDetailsResult
 	err = db.Txn(ctx, func(ctx context.Context, tx *sqlair.TX) error {
-		// TODO: Do this all in one query
-		relations, err := st.getAllRelations(ctx, tx)
+		// Get all non-synthetic relations with their basic info in one query.
+		relations, err := st.getAllNonSyntheticRelationsWithDetails(ctx, tx)
 		if err != nil && !errors.Is(err, sqlair.ErrNoRows) {
 			return errors.Errorf("getting all relations: %w", err)
 		}
 
-		for _, rel := range relations {
-			details, err := st.getRelationDetails(ctx, tx, rel.UUID)
-			if err != nil {
-				return errors.Errorf("getting relation details: %w", err)
+		if len(relations) == 0 {
+			return nil
+		}
+
+		// Extract relation UUIDs for batch queries.
+		relationUUIDs := make([]string, len(relations))
+		for i, rel := range relations {
+			relationUUIDs[i] = rel.UUID.String()
+		}
+
+		// Get all endpoints for these relations in one query.
+		endpointsByRelation, err := st.getRelationEndpointsBatch(ctx, tx, relationUUIDs)
+		if err != nil {
+			return errors.Errorf("getting relation endpoints: %w", err)
+		}
+
+		// Get in-scope unit counts for these relations in one query.
+		inScopeCountsByRelation, err := st.countInScopeRelationsBatch(ctx, tx, relationUUIDs)
+		if err != nil {
+			return errors.Errorf("counting in-scope relations: %w", err)
+		}
+
+		// Assemble the final results.
+		relationsDetails = make([]domainrelation.RelationDetailsResult, len(relations))
+		for i, rel := range relations {
+			relationsDetails[i] = domainrelation.RelationDetailsResult{
+				Life:         rel.Life,
+				UUID:         corerelation.UUID(rel.UUID),
+				ID:           rel.ID,
+				Suspended:    rel.Suspended,
+				Endpoints:    endpointsByRelation[rel.UUID.String()],
+				InScopeUnits: inScopeCountsByRelation[rel.UUID.String()],
```

Comment:
We should have a test that ensures that we **always** get the relation uuid in the maps or we should be more defensive and return an error or log a warning about missing data.